sources/packages/api/src/route-controllers/institution/institution.controller.ts

```diff
@@ -0,0 +1,40 @@
+import {
+  Body,
+  Controller,
+  Get,
+  Post,
+  UnprocessableEntityException,
+} from "@nestjs/common";
+
+import { Public } from "../../auth/decorators/public.decorator";
+
+import { InstitutionService, UserService } from "../../services";
+import { CreateInstitutionDto } from "./models/institution.dto";
+import { UserToken } from "../../auth/decorators/userToken.decorator";
+import { IUserToken } from "../../auth/userToken.interface";
+import BaseController from "../BaseController";
+
+@Controller("institution")
+export class InstitutionController extends BaseController {
+  constructor(
+    private readonly userService: UserService,
+    private readonly institutionService: InstitutionService,
+  ) {
+    super();
+  }
+
+  @Post()
+  async create(
+    @Body() payload: CreateInstitutionDto,
+    @UserToken() userToken: IUserToken,
+  ) {
+    // Check user exists or not
+    const existingUser = await this.userService.getUser(userToken.userName);
+    if (existingUser) {
+      throw new UnprocessableEntityException("Institution User already exists");
+    }
+
+    // Save student
+    return this.institutionService.createInstitution(userToken, payload);
```

Comment:
It seems that the entire Institution entity is returned here, if it is, please do not return it.

Reply:
hmmm what is the problem in returning?  - Data security? Serialization overhead?Should we return void?
Student Controller returns the whole thing too?

Reply:
The main concern is that we are going to be returning an entire object from the other layer when we have already talked about returning only using Dto, so it falls under security also.
If the student controller is doing it we need to change it, in my option.

Reply:
Returnign void